### PR TITLE
Explicitly define the apps and services we build via build.sh

### PR DESCRIPTION
**Pull request by amcmanus**
See https://github.com/votingworks/vxsuite-complete-system/issues/315 for details.

Rather than relying on various directories being inspected, this now defines the apps and services to build. 

I also reorganized some of the script for clarity and to make the usage possible to get without first installing dependencies.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -22,44 +22,72 @@ set -euo pipefail
...
+      exit 1
+    else
+      echo "✘ unknown app: ${arg}" >&2
+      usage >&2
+      exit 1
+    fi
+  done
+fi
+
+# Install linux dependencies for all defined apps before we build
```
[the remark this answers is not in the document]
Comment (amcmanus) [reply build.sh:67]: Great question. Mainly, I didn't wanna mess with it in the context of this PR to keep it focused and address the original issue. That said, we've got some implicit assumptions within apps that rely on another app having already built a dependency, so it's simpler (for now) to just build these dependencies all at once. We also don't call `build.sh` from `setup-machine.sh` anymore to make building images faster, so this ensures everything is already built for whatever specific image we're creating.

File: build.sh [hunk 2]
```diff
@@ -22,44 +22,72 @@ set -euo pipefail
...
-ALL_APPS=()
-
-# Install linux dependencies for all apps
-for app in ${DIR}/vxsuite/apps/*; do
-  if [ -d "${app}" ]; then
-    tmp_dir=$(basename "${app}")
-    if [ -d "${app}/frontend" ]; then
-      make -C "${DIR}/vxsuite/apps/${tmp_dir}/frontend" install
+# Define vxsuite apps that can be built, along with the expected path prefix
+ALL_APPS=(admin central-scan mark scan)
```
Comment (eventualbuddha) [inline build.sh:26]: @kshen0 just added `mark-scan`, right?

```suggestion
ALL_APPS=(admin central-scan mark scan mark-scan)
```

**Review by eventualbuddha — APPROVED** [review]

File: build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahkagan) [reply build.sh:67]: Sounds good

File: build.sh (hunk 2 above)
Comment (amcmanus) [reply build.sh:26]: `mark-scan` was excluded in an earlier workaround, so I left it excluded. If @kshen0 can confirm it's ready to be included in this build process, happy to update.

**Pull request merged**